hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastVersionLocator.java

```diff
@@ -16,105 +16,150 @@
-        if (Version.of(version).isGreaterOrEqual(Versions.V5_0)) {
-            files.put(Artifact.SQL_JAR, locateArtifact(Artifact.SQL_JAR, version));
+            if (!localCopy.exists()) {
+                downloadArtifact(version);
+            }
+            return localCopy;
         }
-        if (enterprise) {
-            files.put(Artifact.EE_JAR, locateArtifact(Artifact.EE_JAR, version));
-        }
-        return files;
-    }
 
-    private static File locateArtifact(Artifact artifact, String version) {
-        String path = format(artifact.path, version);
-        File localCopy = new File(LOCAL_M2_REPOSITORY_PREFIX + path);
-        if (!localCopy.exists()) {
-            downloadArtifact(artifact, version);
+        private static String getMvn() {
+            if (OsHelper.isWindows()) {
+                return "mvn.cmd";
+            } else {
+                final String mvn = "mvn";
+
+                if (!new File(mvn).exists() && OsHelper.isMac()) {
+                    // Eclipse doesn't properly read the $PATH, so hardcode another location -
+                    // https://stackoverflow.com/q/76866453
+                    return "/opt/homebrew/bin/mvn";
+                } else {
+                    return mvn;
+                }
+            }
         }
-        return localCopy;
-    }
 
-    private static void downloadArtifact(Artifact artifact, String version) {
-        logWarningForArtifactDownload(artifact, version);
-        ProcessBuilder builder = new ProcessBuilder(buildMavenCommand(artifact, version).split(" "))
-                .inheritIO();
-        try {
-            Process process = builder.start();
-            boolean successful = process.waitFor(120, TimeUnit.SECONDS);
-            checkState(successful, "Maven dependency:get timed out");
-            checkState(process.exitValue() == 0, "Maven dependency:get failed");
-        } catch (InterruptedException | IOException e) {
-            throw new RuntimeException("Problem in invoking Maven dependency:get " + artifact + ":" + version , e);
+        private void downloadArtifact(final String version) {
+            // It's also possible to download this via Aether but I was unable to get this working
+            final ProcessBuilder builder = new ProcessBuilder(buildMavenCommand(version).toArray(String[]::new)).inheritIO();
+            try {
+                final Process process = builder.start();
+                final boolean successful = process.waitFor(2, TimeUnit.MINUTES);
+                checkState(successful, "Maven dependency:get timed out");
+                checkState(process.exitValue() == 0, "Maven dependency:get failed");
+            } catch (InterruptedException | IOException e) {
+                throw new RuntimeException("Problem in invoking Maven dependency:get " + toString() + ":" + version, e);
+            }
         }
-    }
 
-    private static void logWarningForArtifactDownload(Artifact artifact, String version) {
```

Comment:
> This warning about artifact downloads has been removed - was this intentional? Seems like a useful warning for local usage.

I removed it because I don't understand why it's there (risk of invoking Chesterton's Fence here).

The warning appears just before it runs a `mvn dependency:get`. That will download the version, if not already present, and store it for next time.
So telling you "next time, run the `mvn dependency:get` command it won't redownload it" is just wrong - because it's just done that.
If the Maven command downloaded to some temporary directory it'd make sense, or if it was a `SNAPSHOT` dependency (and then the test command didn't force updates). But it's not. So I don't understand.

Reminds me though, I need to update the EE readme as well.